pkg/commands/compute/deploy.go

```diff
@@ -1115,3 +1111,75 @@ func processService(c *DeployCommand, serviceID string, serviceVersion int, spin
 	spinner.StopMessage(msg)
 	return spinner.Stop()
 }
+
+func getServiceDomain(apiClient api.Interface, serviceID string, serviceVersion int) (string, error) {
+	latestDomains, err := apiClient.ListDomains(&fastly.ListDomainsInput{
+		ServiceID:      serviceID,
+		ServiceVersion: serviceVersion,
+	})
+	if err != nil {
+		return "", err
+	}
+	name := latestDomains[0].Name
+	if segs := strings.Split(name, "*."); len(segs) > 1 {
+		name = segs[1]
+	}
+	return name, nil
+}
+
+// checkingServiceAvailability pings the service URL until either there is a 200
+// OK or if the configured timeout is exceeded.
+func checkingServiceAvailability(serviceURL string, spinner text.Spinner) error {
+	err := spinner.Start()
+	if err != nil {
+		return err
+	}
+	msg := "Checking service availability"
+	spinner.Message(msg + "...")
+
+	timeout := time.After(2 * time.Minute)
+	ticker := time.NewTicker(1 * time.Second)
+	defer func() { ticker.Stop() }()
+
+	// Keep trying until we're timed out, got a result or got an error
+	for {
+		select {
+		case <-timeout:
+			spinner.StopFailMessage(msg)
+			spinErr := spinner.StopFail()
+			if spinErr != nil {
+				return spinErr
+			}
+			return errors.New("service unavailable")
+		case <-ticker.C:
+			ok, err := pingServiceURL(serviceURL)
+			if err != nil {
+				spinner.StopFailMessage(msg)
+				spinErr := spinner.StopFail()
+				if spinErr != nil {
+					return spinErr
+				}
+				return err
+			} else if ok {
+				spinner.StopMessage(msg)
+				return spinner.Stop()
+			}
+			// Service not available, and no error, so jump back to top of loop
+		}
+	}
+}
+
+func pingServiceURL(serviceURL string) (ok bool, err error) {
+	// gosec flagged this:
+	// G107 (CWE-88): Potential HTTP request made with variable url
+	// Disabling as we trust the source of the variable.
+	// #nosec
+	resp, err := http.Get(serviceURL)
+	if err != nil {
+		return false, err
+	}
+	if resp.StatusCode == http.StatusOK {
```

Comment:
We don't control the service, which means we can't know what status code the root path should be returning -- a 400 could be a valid response for example -- is there something else we could be using to check service availability?

Reply:
Hmm. Good point!

Reply:
Would checking for something that wasn't a `500 Internal Server Error` be sufficient?